src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -944,16 +949,24 @@ TableIdListSelectClauses(List *tableIdList, List *selectClauseList)
 	foreach(selectClauseCell, selectClauseList)
 	{
 		Node *selectClause = (Node *) lfirst(selectClauseCell);
-		List *selectColumnList = pull_var_clause_default(selectClause);
-
-		Var *selectColumn = (Var *) linitial(selectColumnList);
-		int selectClauseTableId = (int) selectColumn->varno;
 
-		bool tableIdListMember = list_member_int(tableIdList, selectClauseTableId);
-		if (tableIdListMember)
+		List *selectColumnList = pull_var_clause_default(selectClause);
+		if (list_length(selectColumnList) == 0)
 		{
+			/* filter is a constant, e.g. false or 1=0, always include it */
 			tableSelectClauseList = lappend(tableSelectClauseList, selectClause);
 		}
+		else
+		{
+			Var *selectColumn = (Var *) linitial(selectColumnList);
```

Comment:
In that case it would be a join clause and it would not appear in a MultiSelect node.